postqa/jsonshim.py

```diff
@@ -9,33 +9,74 @@
 install_aliases()  # NOQA
 
 
-def shim_validate_drp(vdrp_json, accepted_metrics):
-    """Convert JSON structure from validate DRP into the JSON expected by
-    SQuaSH's POST /api/jobs/ endpoint.
+def shim_validate_drp(vdrp_json, registered_metrics=[]):
+    """Convert JSON structure from validate DRP into the JSON documents
+    expected by SQuaSH's API.
 
-    This makes a full job document, but only populates the ``measurements``
-    field.
+    Populates the ``blobs`` and ``measurements`` fields of the job document
+    expected by the api/jobs endpoint.
+
+    For metrics not registered in SQuaSH, makes a metric document as
+    expected by the api/metrics endpoint.
     """
-    job_json = {}
 
+    job_json = {
+        'blobs': vdrp_json['blobs']
+    }
+
+    metrics = []
     measurements = []
     for vdrp_measurement_doc in vdrp_json['measurements']:
-        if vdrp_measurement_doc['metric']['name'] not in accepted_metrics:
-            continue
         if vdrp_measurement_doc['value'] is None:
             continue
+        if vdrp_measurement_doc['metric']['name'] not in registered_metrics:
+            metrics.append(shim_metric_definition(vdrp_measurement_doc))
```

Comment:
I propose to defer this issue to another ticket in order to close this one. Opened https://jira.lsstcorp.org/browse/DM-9330